sdk/keyvault/Azure.Security.KeyVault.Keys/src/RecoverDeletedKeyOperation.cs

```diff
@@ -5,34 +5,37 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Azure.Core;
-using Azure.Core.Pipeline;
 
 namespace Azure.Security.KeyVault.Keys
 {
     /// <summary>
     /// A long-running operation for <see cref="KeyClient.StartRecoverDeletedKey(string, CancellationToken)"/> or <see cref="KeyClient.StartRecoverDeletedKeyAsync(string, CancellationToken)"/>.
     /// </summary>
-    public class RecoverDeletedKeyOperation : Operation<KeyVaultKey>
+    public class RecoverDeletedKeyOperation : Operation<KeyVaultKey>, IOperation<KeyVaultKey>
     {
         private static readonly TimeSpan s_defaultPollingInterval = TimeSpan.FromSeconds(2);
 
+        private readonly OperationInternal<KeyVaultKey> _operationInternal;
         private readonly KeyVaultPipeline _pipeline;
         private readonly KeyVaultKey _value;
-        private Response _response;
-        private bool _completed;
 
         internal RecoverDeletedKeyOperation(KeyVaultPipeline pipeline, Response<KeyVaultKey> response)
         {
             _pipeline = pipeline;
             _value = response.Value ?? throw new InvalidOperationException("The response does not contain a value.");
-            _response = response.GetRawResponse();
+            _operationInternal = new(pipeline.Diagnostics, this, response.GetRawResponse())
+            {
+                DefaultPollingInterval = s_defaultPollingInterval
+            };
+
+            _operationInternal.ScopeAttributes.Add("key", Value.Name);
         }
 
         /// <summary> Initializes a new instance of <see cref="RecoverDeletedKeyOperation" /> for mocking. </summary>
         protected RecoverDeletedKeyOperation() {}
 
         /// <inheritdoc/>
-        public override string Id => _value.Id.ToString();
+        public override string Id => Value.Id.ToString();
```

Comment:
I don't think this is going to work for us. For pseudo-LROs Delete* and RecoveredDeleted* that we have across libraries, they always have a value and don't actually get it from the response. These aren't true LROs, but implemented as such because they take a while on the service (the service probably should've defined them as LROs, but now it would be breaking).

Reply:
I believe it's still possible to use the new helper class, though, but I'm not sure if you think it's worth the migration. You could still have the `_value` private field:
```cs
public override string Id => _value.Id.ToString();

public override <Type> Value => _value;

public override bool HasValue => true;
```

However, once you consider the operation "complete", notify the helper class like other operations would do:
```cs
// In UpdateState:
return OperationState<Type>.Success(response, _value); 
```

With this, you can get implementations for every other method/property and make sure they behave like the operations in other clients, but I'll leave it at your discretion.

Reply:
Maybe. I'll evaluate later. But it does make me think of another thing: The default polling interval of 1s should probably be consolidated into a single place we can change later - or is the idea the only other place it is now will go away soon enough?

Reply:
We'll remove the original `OperationHelpers` when migration is complete.